app/models/auth_sources/auth_source_ldap.rb

```diff
@@ -130,6 +131,28 @@ def users_in_group(name)
     ldap_con.user_list(name)
   end
 
+  def test_connection
+    result = {}
+    begin
+      Timeout::timeout(10) do
+        self.ldap_con.valid_user?("")
```

Comment:
Please extend the test method over in theforeman/ldap_fluff, it's a good opportunity to make sure it tests a variety of useful attributes, such as existence of base DNs etc.